content/guides/advanced/testing-with-jest-and-bankrun.md

```diff
@@ -0,0 +1,380 @@
+to the bank by passing them in the `startAnchor` function. There are a few
+things that are different to running tests against the local validator though
+which we will cover in the next section.
+
+### Bankrun differences to the local validator
+
+Because Bankrun directly runs your tests against a local Bank the tests are
+faster, but there are a few differences to the local validator:
+
+#### Airdrops
+
+- Bankrun does not support Airdrops. The standard signer used in the
+  BankrunProvider will be automatically funded with some Sol. If you need
+  another funded account you can create one by passing in an additional account
+  in the `startAnchor` function.
+
+```js
+let secondKeypair: Keypair = new anchor.web3.Keypair();
+
+context = await startAnchor(
+    "",[],
+    [
+    {
+        address: secondKeypair.publicKey,
+        info: {
+        lamports: 1_000_000_000, // 1 SOL equivalent
+        data: Buffer.alloc(0),
+        owner: SYSTEM_PROGRAM_ID,
+        executable: false,
+        },
+    },
+    ]
+);
+provider = new BankrunProvider(context);
+```
+
+#### Confirming transactions
+
+Since Bankrun is directly working on the bank you will not need to confirm your
+transactions. So the `connection.confirmTransaction()` function will not be
+available. You can just leave it out.
+
+#### Getting account data
+
+While you can still use `connection.getAccount` to get the account data of an
+account you can also use the `client.getAccount` function to get the account
+data of an account. This will return a `Promise<Account>`.
+
+```js
+await client.getAccount(playerPDA).then(info => {
+  const decoded = program.coder.accounts.decode(
+    "playerData",
+    Buffer.from(info.data),
+  );
+  console.log("Player account info", JSON.stringify(decoded));
+  expect(decoded).toBeDefined();
+  expect(parseInt(decoded.energy)).toEqual(99);
+});
+```
+
+#### Signing transactions with another keypair
+
+By default when using `program.function.rpc()` the transaction will be
+automatically signed with the `provider.wallet` keypair. If you want to sign the
+transaction with another keypair you can create a second provider and then use
+that one to sign transaction with another keypair.
+
+```js
+let secondKeypair: Keypair = new anchor.web3.Keypair();
+
+let context = await startAnchor(
+"",[],
+[
+    {
+        address: secondKeypair.publicKey,
+        info: {
+        lamports: 1_000_000_000,
+        data: Buffer.alloc(0),
+        owner: SYSTEM_PROGRAM_ID,
+        executable: false,
+        },
+    },
+    ]
+);
+beneficiaryProvider = new BankrunProvider(context);
+beneficiaryProvider.wallet = new NodeWallet(secondKeypair);
+
+secondProgram = new Program<Vesting>(IDL as Vesting, beneficiaryProvider);
+```
+
+### Using Bankrun for native programs
+
+You can also use Bankrun for native programs. The main difference is that you
+use `start` instead of `startAnchor` to start the Bankrun bank. You can then use
+the `client` to interact with the bank.
```

Comment:
Internal linking for SEO :)
```suggestion
You can also use Bankrun for
[native programs](/content/guides/getstarted/intro-to-native-rust.md). The main
difference is that you use `start` instead of `startAnchor` to start the Bankrun
bank. You can then use the `client` to interact with the bank.
```